### ISSUE TITLE
Feature request: rename columns when selecting

### ISSUE THREAD

**Issue by purcell** [BurntSushi/xsv#173]
Since `xsv` is so invaluable for splitting and reformatting big messy CSV files (thanks!), it'd be really useful to be able to rename columns when `select`-ing them out: I deal with some crappy CSV files that have 150-character header names.

At first I was thinking a special syntax for `select` might be the solution, but now I suspect a `rename` subcommand might be the more appropriate option. Not really up to speed with Rust dev yet, so starting off by sharing this idea for feedback.

**Comment by BurntSushi** [comment]
This sounds like a possibly good idea to me. It probably should be a separate rename command. It will need some design work in terms of the syntax supported. It would ideally combine select syntax for specifying which columns to rename, for example.

**Comment by purcell** [comment]
Yeah, I'm thinking something like `1=New name,Code=value,Dupe[3]="Nondupe","Quoted[2]"=something`. There's no need for the `1-4` column range syntax, or `!` exclusion for a pure `rename` command, which makes this simpler than cramming renaming into `select`.

**Comment by BurntSushi** [comment]
A simpler version might be too just keep the same select syntax, which selects some number of columns in a particular order. Then the remaining positional arguments are the new names of the columns, where there are exactly N arguments, where N is the number of fields selected.

**Comment by purcell** [comment]
Ah, yes. I can see that might work. The simple case would then just be `xsv rename "Foo bar" "Fubar"`, which is nice.

I had a look through past issues etc., and what we're talking about here sounds a bit like the `--set-headers` suggestion which was part of #96, but a bit cleaner IMO.

**Comment by kacrouse** [comment]
+1 for this feature! Definitely have had the need for this a few times.

**Comment by jlumbroso** [comment]
@BurntSushi @purcell @kacrouse I had to merge/join CSV files today, and I just stumbled upon `xsv` and it's a really neat project!

I really needed to be able to rename columns so I tried to figure out what would be the best way to contribute a solution. I implemented what was discussed in this thread, specifically the variant suggested by @BurntSushi. Let me know if this hits the spot!

**Comment by jillesvangurp** [comment]
a sql like syntax for this so something like `xsv select "personName as name,user_id as id" users.csv`

**Comment by BurntSushi** [comment]
`xsv` is never going to get a SQL-like syntax. If you want SQL, load the data into SQLite or some other relation database.

**Comment by jillesvangurp** [comment]
I appreciate that you don't want to support the full sql syntax. I already use csvsql (comes with csvkit), which indeed uses sqlite. My main issue with that is that it is kind of slow compared to xsv, which is awesome.

The reason I proposed this `AS` syntax is because it is an elegant solution with minimal impact that is easy to understand and probably also to implement and pretty.
